pynicotine/portmapper.py

```diff
@@ -126,17 +127,25 @@ def _request_port_mapping(self, public_port, private_port, lease_duration):
 
         with socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP) as sock:
             sock.bind((self.local_ip_address, 0))
-            sock.settimeout(self.REQUEST_TIMEOUT)
 
             request = self.PortmapRequest(public_port, private_port, lease_duration)
-            request.sendto(sock, (self._gateway_address, self.REQUEST_PORT))
 
-            try:
-                response = self.PortmapResponse(message=sock.recv(16))
-                return response.result
+            timeout = self.REQUEST_INIT_TIMEOUT
+            i = 1
+            while i <= self.REQUEST_ATTEMPTS:
+                sock.settimeout(timeout)
+                request.sendto(sock, (self._gateway_address, self.REQUEST_PORT))
```

Comment:
suggest adding an argument to log the number of the current retry attempt `i` in `sendto()` instead of here...

`request.sendto(sock, (self._gateway_address, self.REQUEST_PORT, i))`


```
        def sendto(self, sock, addr, num_attempt):

            msg = bytes(self)
            sock.sendto(msg, addr)

            log.add_debug(f"NAT-PMP: Portmap request attempt {num_attempt} of {self.REQUEST_ATTEMPTS}: {msg}")
```

Then additional log lines would not be needed, that saves on the amount of log lines.